### PR TITLE
Sad path take 3

### PR DESCRIPTION
**Pull request by EricLarson2020**
-Created empty dropdown for advanced search
-Added sad test for if no entries filled out

### CODE REVIEW

File: app/controllers/search/results_controller.rb [hunk 1]
```diff
@@ -1,9 +1,13 @@
 class Search::ResultsController < ApplicationController
 
   def index
-    @obituaries_api = ObituarySearch.new.advanced_query(params) || []
+    if params[:name].empty? && params[:location].empty? && params[:age].empty? && params[:year].empty?
```
Comment (EricLarson2020) [inline app/controllers/search/results_controller.rb:4]: If no entries are given redirects back to advanced search page

File: app/facades/obituary_search.rb [hunk 2]
```diff
@@ -55,15 +57,16 @@ def advanced_query_local(params)
   end
 
   def parse_params(params)
+
     query_params = {}
     params.each do |attribute, query|
       if attribute == "name" && query != ""
         query_params[:name] = query
-      elsif attribute == "year" && query != "0"
+      elsif attribute == "year" && query != ""
```
Comment (EricLarson2020) [inline app/facades/obituary_search.rb:65]: Updates to match with empty dropdown

File: app/views/search/advanced/index.html.erb [hunk 3]
```diff
@@ -22,14 +22,14 @@
           <div class="form-group-search row" id="age-select">
             <div class="col col-form-label-lg" >
               <%= label_tag :age %>
-              <%= select_tag :age, options_for_select((0..150), { }), {class: 'custom-select' } %>
+              <%= select_tag :age, options_for_select((0..150), { }), class: 'custom-select', include_blank: true %>
```
Comment (EricLarson2020) [inline app/views/search/advanced/index.html.erb:25]: Makes empty dropdown

File: app/views/application/_browsing_index.html.erb [hunk 4]
```diff
@@ -1,6 +1,7 @@
 <div class="container">
-<% cache(action: "index", action_suffix: "obituaries_index") do %>
+
   <% @obituaries.each do |obit| %>
+  <% cache(obit) do %>
```
Comment (jpc20) [inline app/views/application/_browsing_index.html.erb:4]: Does this work on your local host?

File: app/views/application/_browsing_index.html.erb [hunk 5]
```diff
@@ -1,6 +1,7 @@
 <div class="container">
-<% cache(action: "index", action_suffix: "obituaries_index") do %>
+
   <% @obituaries.each do |obit| %>
```
Comment (EricLarson2020) [inline app/views/application/_browsing_index.html.erb:3]: hoping that this associates the cache with each obit object, so when it sees a new one it knows to update the cache

File: app/views/application/_browsing_index.html.erb (hunk 4 above)
Comment (EricLarson2020) [reply app/views/application/_browsing_index.html.erb:4]: yes

Comment (EricLarson2020) [reply app/views/application/_browsing_index.html.erb:4]: with rails dev:cache
Development mode is now being cached.

Comment (jpc20) [reply app/views/application/_browsing_index.html.erb:4]: Okay sounds good.

File: app/views/search/advanced/index.html.erb (hunk 3 above)
Comment (jpc20) [reply app/views/search/advanced/index.html.erb:25]: Nice fix.

File: app/controllers/search/results_controller.rb (hunk 1 above)
Comment (sagemlee) [reply app/controllers/search/results_controller.rb:4]: Hope this fixes it! Thanks for working on it Eric.

**Pull request merged**